lmdeploy/pytorch/engine/executor/ray_executor.py

```diff
@@ -185,29 +194,36 @@ def __init__(self,
                          model_config=model_config,
                          cache_config=cache_config,
                          backend_config=backend_config,
+                         dist_config=dist_config,
                          tokenizer=tokenizer,
-                         dp=dp,
-                         tp=tp,
                          adapters=adapters,
                          device_type=device_type)
 
-        self.world_size = tp * dp
+        self.dp_rank = dist_config.dp_rank
         device_ctx = DeviceContext(device_type)
         with get_device_manager().context(device_ctx):
             logger.info('Init ray cluster.')
-            placement_group = init_ray_cluster(self.world_size)
+            ray_world_size = self.world_size
+            if self.dp > 1:
+                ray_world_size = 1
+            placement_group = init_ray_cluster(ray_world_size)
         self.placement_group = placement_group
-        self.master_addr = _get_master_addr()
-        self.master_port = find_available_port()
-        setup_master_addr(self.master_addr, self.master_port)
+
+        if self.dp == 1:
+            self.master_addr = _get_master_addr()
+            self.master_port = _get_master_port()
+        else:
+            self.master_addr = os.environ.get('LMDEPLOY_DP_MASTER_ADDR', None)
```

Comment:
Can we use `LMDEPLOY_MASTER_ADDR` and `LMDEPLOY_MASTER_PORT`?

Reply:
It is op for me, but vllm use a dp special addr and port, should we follow?

Reply:
OK